src/Firebase/Auth.php

```diff
@@ -437,6 +437,56 @@ public function verifyPassword($email, $password): UserRecord
         return $this->getUserRecordFromResponse($response);
     }
 
+    /**
+     * Verifies password reset code.
+     *
+     * 3 errors code possible if the oobCode is invalid:
+     * - OPERATION_NOT_ALLOWED: Password sign-in is disabled for this project.
+     * - EXPIRED_OOB_CODE: The action code has expired.
+     * - INVALID_OOB_CODE: The action code is invalid. This can happen if the code is malformed, expired, or has already been used.
+     *
+     * @param string $oobCode the email action code sent to the user's email for resetting the password
+     *
+     * @throws Exception\AuthException
+     * @throws Exception\FirebaseException
+     *
+     * @return mixed
+     *
+     * @see https://firebase.google.com/docs/reference/rest/auth#section-verify-password-reset-code
+     */
+    public function verifyPasswordResetCode(string $oobCode)
+    {
+        $response = $this->client->verifyPasswordResetCode($oobCode);
+
+        return JSON::decode((string) $response->getBody(), true);
+    }
+
+    /**
+     * Confirm the reset password.
+     *
+     * 4 errors code possible:
+     * - OPERATION_NOT_ALLOWED: Password sign-in is disabled for this project.
+     * - EXPIRED_OOB_CODE: The action code has expired.
+     * - INVALID_OOB_CODE: The action code is invalid. This can happen if the code is malformed, expired, or has already been used.
+     * - USER_DISABLED: The user account has been disabled by an administrator.
+     *
+     * @param string $oobCode the email action code sent to the user's email for resetting the password
+     * @param string $newPassword the user's new password
+     *
+     * @throws Exception\AuthException
+     * @throws Exception\FirebaseException
+     *
+     * @return mixed
+     *
+     * @see https://firebase.google.com/docs/reference/rest/auth#section-confirm-reset-password
+     */
+    public function confirmPasswordReset(string $oobCode, string $newPassword)
```

Comment:
For consistency, I would handle the password similar to the `verifyPassword()` method:

```php
/**
 * ...
 * @param Email|string $email
 * @param ClearTextPassword|string $newPassword
 *
 * @throws InvalidArgumentException if the given password is invalid
 * ...
 */
public function confirmPasswordReset(string $oobCode, $newPassword)
{
    $newPassword = $newPassword instanceof ClearTextPassword ? $newPassword : new ClearTextPassword($$newPassword);

    $response = $this->client->confirmPasswordReset($oobCode, (string) $newPassword);
    // ...
}
```